docs/core/event_handler/bedrock_agents.md

```diff
@@ -0,0 +1,291 @@
+* **Lambda permissions** allowing Amazon Bedrock to invoke it
+
+=== "Service Role Example using AWS CloudFormation"
+	You'll need this role when [creating an Agent in the AWS Console](#walkthrough-for-creating-the-agent).
+
+  	```yaml hl_lines="14-17 24-29 34"
+  	--8<-- "examples/event_handler_bedrock_agents/sam/bedrock_service_role.yaml"
+  	```
+
+	1. Check the [supported foundational models](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-supported.html){target="_blank"}
+	2. You need the role ARN when creating the Agent for Amazon Bedrock
+
+=== "Lambda permissions using AWS Serverless Application Model (SAM)"
+	```yaml hl_lines="27-33"
+	--8<-- "examples/event_handler_bedrock_agents/sam/template.yaml"
+	```
+
+	1. Amazon Bedrock needs permissions to invoke this Lambda function
+
+=== "Using AWS Cloud Developer Kit (CDK)"
+	Use the [Generative AI CDK constructs](https://awslabs.github.io/generative-ai-cdk-constructs/src/cdk-lib/bedrock/#agents){target="_blank"} to create your Agent with [AWS CDK](https://aws.amazon.com/cdk/){target="_blank"}.
+	These constructs abstract all the underlying permission setup and bundling of your Lambda function.
+
+	```python
+    --8<-- "examples/event_handler_bedrock_agents/cdk/bedrock_agent_stack.py"
+	```
+
+	1. The path to your Lambda function handler
+	2. The path to the OpenAPI schema describing your API
+
+### Your first Agent
+
+To create an Agent for Amazon Bedrock, use the `BedrockAgentResolver` to annotate your actions.
+This is similar to the way [all the other Event Handler](api_gateway.md) resolvers work.
+
+=== "Lambda handler"
+
+	The resolvers used by Agents for Amazon Bedrock are compatible with the full suite of Powertools for AWS Lambda utilities.
+	This includes [Logger](../logger.md), [Metrics](../metrics.md) and [Tracer](../tracer.md).
+
+    ```python hl_lines="4 9 12 21"
+    --8<-- "examples/event_handler_bedrock_agents/src/getting_started.py"
+    ```
+
+	1. `description` is a **required** field that should contain a human readable description of your action
+	2. We take care of **parsing**, **validating**, **routing** and **responding** to the request.
+
+=== "OpenAPI schema"
+
+	Powertools for AWS Lambda [generates this automatically](#generating-openapi-schemas) from the Lambda handler.
+
+	```json
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started_schema.json"
+	```
+
+=== "Input payload"
+
+	```json hl_lines="4 6 13"
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started.json"
+	```
+
+=== "Output payload"
+
+	```json hl_lines="10"
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started_output.json"
+	```
+
+!!! note "It's required to include a `description` for each API endpoint and input parameter. This will improve the understanding Amazon Bedrock has of your actions."
+
+### Validating input and output
+
+You can define the expected format for incoming data and responses by using type annotations.
```

Comment:
Add more info here, as the code is quite terse coming from the first example. As suggested in the last feedback session, this might be the first time someone sees Pydantic models and all annotations here.

Give context.

Reply:
I acted on the session feedback by adding extensive code annotations, but I understand that we might need more. I've changed it now to more explicit about what's going on here.